### DISCUSSION TITLE
[Bug] When changing the start button style, hovering over the start button causes it to change to what it was before

### DISCUSSION THREAD

**Discussion by sdw4938** [valinet/ExplorerPatcher#968] [General]
When changing the start button style, hovering over the start button causes it to change to what it was before. Moving your cursor away allows makes the start button use the new style that you selected.

Steps:

1. Open ExplorerPatcher's properties window
2. Click "Taskbar"
3. Change the start button style
4. Hovering over the start button causes it to change to the old one.

Restarting explorer.exe fixes the problem.
The video might be laggy in some places because of explorer starting

https://user-images.githubusercontent.com/76626288/156258999-9215d092-302d-45c7-885f-7d419f1cd116.mp4

**Comment by valinet** [comment]
Yeah, I know, there's a cache somewhere, obviously. Haven't looked into it much further, supposing that users choose a style and never bother with it. On some system it works it works fine. I thought it's better to leave it as is rather than put an * near it, requiring a File Explorer restart on any change. Yeah, if you want to take a look at it, figure out what the missing piece of the puzzle is, be my guest.

**Reply by sdw4938** [reply]
What is a good place to start?

**Reply by valinet** [reply]
What do you mean? In EP, the part responsible for this is around `explorer_DrawThemeBackground` function in `dllmain.c`.

**Comment by plofhaan** [comment]
This is **PRECISELY** why I love Open-Shell so much. :-D
